Milestone 5- Project completion

Stephanie041996 opened the pull request:
For this last milestone, the goal was to tie the calculations module with the components  andadd the interaction to the buttons
-Added class component and Initialize its state in the constructor 
-Implement a render function that should display Display and ButtonPanel.
-Implement the handleClick
-implement handleClick to ButtonPanel and Button

Stephanie041996 committed:
Add class component

Stephanie041996 committed:
Implement render function

Stephanie041996 committed:
Add handleclick function

Stephanie041996 committed:
Pass total and next to Display and handleClick as a clickHandler

Stephanie041996 committed:
Add handleClick to ButtonPanel

Stephanie041996 committed:
Add handleClick to Button.js

Stephanie041996 committed:
Add style and screenshot

Stephanie041996 committed:
Fix linter errors

Stephanie041996 committed:
Fix errors

Stephanie041996 committed:
Fix prop error

george-shammar commented on src/components/App.js:
```diff
@@ -1,17 +1,40 @@
+/* eslint-disable react/no-access-state-in-setstate */
```
- [ ] Please don't disbale eslint. Please, let's try to resolve the errors.

- [ ] Please let's fix the bug in the `+/-` operator. It's currently outputing the negative square of a number. eg. pressing `8` and `+/-` gives `-64`.
- [ ] Decimal point is also not working. Please let's fix it.

Stephanie041996 replied:
thanks for the review I have fixed the buttons not working. I think they work fine now. but I'm not able to find solution for the linter error , it is saying to use callbacks and I can't seem to find a way to do this simply with no hooks as I have not yet learnt hooks. if you have working code for the same implementation I used that would be very helpful. I have spent hours and was unable to resolve.

george-shammar reviewed (changes requested):
Hi @Stephanie041996 , 

Good job so far!

:heavy_check_mark: You have implemented almost all the required features and your repo looks professional.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Stephanie041996 committed:
Fix decimal and +/- for review

rukundoeric reviewed (changes requested):
Hi @Stephanie041996, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [x] As it is mentioned in the previous review, `+/-` this operator still has the same issue. it is still outputting the negative square of a number. [Video](https://res.cloudinary.com/newpoint/video/upload/v1629841886/My%20videos/scrnli_8_24_2021_11-43-03_PM_pcls8f.webm)
  - [ ] And also, decimal points are still not working. kindly fix it. [Video](https://res.cloudinary.com/newpoint/video/upload/v1629841886/My%20videos/scrnli_8_24_2021_11-43-03_PM_pcls8f.webm)
  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

Stephanie041996 commented:
> Hi @Stephanie041996,
> 
> _Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
> 
> ## Highlights
> * Follows git workflow.  💯
> * Codebase looks clean and professional. 💯
> * Addressed some of the issues raised from the previous review. 💯
> 
> ## Required Changes ♻️
> * [x]  As it is mentioned in the previous review, `+/-` this operator still has the same issue. it is still outputting the negative square of a number. [Video](https://res.cloudinary.com/newpoint/video/upload/v1629841886/My%20videos/scrnli_8_24_2021_11-43-03_PM_pcls8f.webm)
> * [x]  And also, decimal points are still not working. kindly fix it. [Video](https://res.cloudinary.com/newpoint/video/upload/v1629841886/My%20videos/scrnli_8_24_2021_11-43-03_PM_pcls8f.webm)
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> * N/A
> 
> Cheers and Happy coding!👏👏👏
> 
> **Feel free to leave any questions or comments in the PR thread if something is not 100% clear.**
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

hie . the errors where fixed and now works fine if you clone. App..I did forget to push and push few mins after review request so please have another look and let me know if now working

rukundoeric reviewed (approved):
Hi @Stephanie041996,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
